### PR TITLE
New changes: More portable and works with CentOS 6.6, curl 7.19.7; New config parser

### PR DESCRIPTION
**Pull request by Rp70**
Hi,

I love to submit this new PR. The primary purpose was making it works under CentOS 6.6 so it will be more portable. On CentOS 6.6 Final (Desktop version), the script throws some errors related to:
- [realpath](https://mandoc.dev/realpath): command not found. I use a trick with cd & [pwd](https://mandoc.dev/pwd) which are always available in most systems.
- [dig](https://mandoc.dev/dig): command not found. This binary is also available by default. I use curl with checkip.amazonaws.com instead. checkip.amazonaws.com is hosted by Amazon AWS so I think it's very stable for lots of connections rather than other checkip URLs.
- [cut](https://mandoc.dev/cut): it parsed updateTarget and recordType incorrectly. So I came up with a new config parser which I think it's simpler and works even order of configs changed. It just like the way some Bash script parse its parameters. This way, I think it's easier to write a config writer and we still use the current config format. I tried to re-produce the weird behavour upon opening this PR but can't. I fixed a week ago on CentOS 6.6 without having time to commit until now.
- [curl](https://mandoc.dev/curl#OPTIONS): curl 7.19.7 doesn't support `-H @-` until 7.55.0
> Starting in 7.55.0, this option can take an argument in @filename style, which then adds a header for each line in the input file. Using @- will make curl read the header file from stdin

I tested the script on Docker containers which run:
- Ubuntu 18.04.2 LTS
- Ubuntu 20.04 LTS
- CentOS Linux release 7.6.1810
- CentOS 6.6 (Final)

 Let me know if you have any questions.

Thank you!

### CODE REVIEW

**Comment by hyecheol123** [comment]
Thanks for the PR, and sorry for being late. As soon as I finish my Finals, I will take a look.

File: cloudflare_dynamic_ip_update.bash [hunk 1]
```diff
@@ -125,14 +125,13 @@ count=0
...
         echo "record IP needs to be updated for "${recordName[count]}" with recordType "${recordType[count]}
-        content=$(
-            printf '%b' "$curlHeader" |
-            curl -s -X PUT "https://api.cloudflare.com/client/v4/zones/$zoneID/dns_records/${dnsID[count]}" \
-                 --data '{"type":"'"${recordType[count]}"'","name":"'"${recordName[count]}"'","content":"'"$currentIP"'","proxied":'${recordProxied[count]}'}' \
-                 -H @-
-        )
-
-        if [ $(echo $content | jq '.success') = "true" ]; then
+        content=$(eval $(cat <<CMD
```
Comment (hyecheol123) [inline cloudflare_dynamic_ip_update.bash:128]: May I ask you what the `CMD` does?

Comment (hyecheol123) [reply cloudflare_dynamic_ip_update.bash:128]: @Rp70

Comment (Rp70) [reply cloudflare_dynamic_ip_update.bash:128]: @hyecheol123 `CMD` here is just a Heredoc delimiter. Here is for an example: https://linuxize.com/post/bash-heredoc/#basic-heredoc-examples .  You can go with EOF, EOL or anything you want (may be).

**Pull request merged**